frontend/catalyst/qfunc.py

```diff
@@ -206,13 +247,18 @@ def processing_fn(results):
         return dynamic_one_shot_partial(qnode)
 
     single_shot_qnode = transform_to_single_shot(qnode)
+    if mcm_config is not None:
+        single_shot_qnode.execute_kwargs["mcm_config"] = mcm_config
+    single_shot_qnode._dynamic_one_shot_called = True
     dev = qnode.device
-    single_shot = 1 if isinstance(dev, qml.devices.LegacyDevice) else qml.measurements.Shots(1)
-    single_name = dev.short_name if isinstance(dev, qml.devices.LegacyDevice) else dev.name
-    single_shot_qnode.device = qml.device(single_name, wires=dev.wires, shots=single_shot)
-    total_shots = (
-        dev.shots if isinstance(qnode.device, qml.devices.LegacyDevice) else dev.shots.total_shots
-    )
+    total_shots = _get_device_shots(dev)
+
+    new_dev = copy(dev)
+    if isinstance(new_dev, qml.devices.LegacyDevice):
+        new_dev.shots = 1
```

Comment:
Codecov says that this line is not covered, but it should be covered since the frontend tests use `lightning.kokkos` as well. Anyone have any ideas why this might be happening?

Reply:
Looks good now, doesn't it?

Reply:
Sadly not. I may have to add a `# pragma: no cover`

Reply:
It's not blocking check so it's possible to merge without as well if you think the line should be covered but for some reason isn't. `# pragma: nocover` should ideally be used for lines that we explicitly don't want to cover.